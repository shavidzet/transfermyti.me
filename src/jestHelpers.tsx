/* eslint-disable import/no-extraneous-dependencies,@typescript-eslint/no-explicit-any,no-console */
import React from "react";
import {
  RenderOptions,
  RenderResult,
  render as testRender,
} from "@testing-library/react";
import { ThemeProvider } from "emotion-theming";
import { theme } from "./components/emotion";

// These are exported for convenience (so we don't have to add an additional import
// to the test file).
export { createEvent, fireEvent, RenderResult } from "@testing-library/react";
export { theme } from "./components/emotion";

/**
 * Renders the specified component with react-testing-library wrapped in a theme
 * provider so we don't have to do it explicitly for each component.
 */
export function render(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">,
): RenderResult {
  return testRender(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, options);
}

type LogMethod = "log" | "warn" | "error";

export class ConsoleToggle {
  private consoleMethod: (...args: any[]) => void = console.warn;
  private readonly logMethod: LogMethod;

  public constructor(logMethod: LogMethod) {
    this.logMethod = logMethod;
  }

  public onBefore(): void {
    this.consoleMethod = console[this.logMethod];
    console[this.logMethod] = jest.fn();
  }

  public onAfter(): void {
    console[this.logMethod] = this.consoleMethod;
  }
}
