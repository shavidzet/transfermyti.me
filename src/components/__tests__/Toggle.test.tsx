import React from "react";
import { fireEvent, render, RenderResult } from "~/jestHelpers";
import Toggle from "../Toggle";

const setup = (
  propOverrides: any = {},
): { props: any; wrapper: RenderResult } => {
  const props = {
    isToggled: false,
    onToggle: jest.fn(),
    ...propOverrides,
  };

  const wrapper = render(<Toggle {...props} />);

  return { props, wrapper };
};

describe("the <Toggle> component", () => {
  test("matches its snapshot when props.isToggled = true", () => {
    const { wrapper } = setup({ isToggled: true });

    expect(wrapper.getByRole("switch")).toMatchSnapshot();
  });

  test("matches its snapshot when props.isToggled = false", () => {
    const { wrapper } = setup({ isToggled: false });

    expect(wrapper.getByRole("switch")).toMatchSnapshot();
  });

  test("fires props.onToggle when toggled", () => {
    const { wrapper, props } = setup({ isToggled: false });
    fireEvent.click(wrapper.getByRole("switch"));

    expect(props.onToggle).toHaveBeenCalled();
  });
});
