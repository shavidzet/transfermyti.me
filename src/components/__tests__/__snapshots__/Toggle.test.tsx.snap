// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the <Toggle> component matches its snapshot when props.isToggled = false 1`] = `
.emotion-1 {
  border: none;
  border-radius: 2rem;
  display: inline-block;
  height: 2rem;
  width: 4.5rem;
  overflow: hidden;
  position: relative;
  -webkit-transition: background 0.1s ease-in-out;
  transition: background 0.1s ease-in-out;
  box-shadow: 0 1px 5px 0 rgba(0,0,0,0.2), 0 2px 2px 0 rgba(0,0,0,0.14), 0 3px 1px -2px rgba(0,0,0,0.12);
}

.emotion-0 {
  border-radius: 50%;
  display: inline-block;
  pointer-events: none;
  position: absolute;
  text-transform: uppercase;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  background: rgb(88,88,88);
  box-shadow: 0 1px 5px 0 rgba(0,0,0,0.2), 0 2px 2px 0 rgba(0,0,0,0.14), 0 3px 1px -2px rgba(0,0,0,0.12);
  font-size: 1rem;
  font-weight: 600;
  left: 0.225rem;
  top: 0.2rem;
  height: 1.6rem;
  width: 1.6rem;
}

.emotion-0::before,
.emotion-0::after {
  display: block;
  position: absolute;
}

.emotion-0::before,
.emotion-0::after {
  color: rgb(30,120,161);
  line-height: 1.7rem;
}

.emotion-0::after {
  content: "NO";
  right: -2.1375rem;
}

<button
  aria-checked="false"
  class="emotion-1"
  role="switch"
>
  <span
    class="emotion-0"
  />
</button>
`;

exports[`the <Toggle> component matches its snapshot when props.isToggled = true 1`] = `
.emotion-1 {
  border: none;
  border-radius: 2rem;
  display: inline-block;
  height: 2rem;
  width: 4.5rem;
  overflow: hidden;
  position: relative;
  -webkit-transition: background 0.1s ease-in-out;
  transition: background 0.1s ease-in-out;
  box-shadow: 0 1px 5px 0 rgba(0,0,0,0.2), 0 2px 2px 0 rgba(0,0,0,0.14), 0 3px 1px -2px rgba(0,0,0,0.12);
}

.emotion-0 {
  border-radius: 50%;
  display: inline-block;
  pointer-events: none;
  position: absolute;
  text-transform: uppercase;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  background: rgb(30,120,161);
  box-shadow: 0 1px 5px 0 rgba(0,0,0,0.2), 0 2px 2px 0 rgba(0,0,0,0.14), 0 3px 1px -2px rgba(0,0,0,0.12);
  font-size: 1rem;
  font-weight: 600;
  left: 2.75rem;
  top: 0.2rem;
  height: 1.6rem;
  width: 1.6rem;
}

.emotion-0::before,
.emotion-0::after {
  display: block;
  position: absolute;
}

.emotion-0::before,
.emotion-0::after {
  color: rgb(30,120,161);
  line-height: 1.7rem;
}

.emotion-0::before {
  content: "YES";
  left: -2.25rem;
}

<button
  aria-checked="true"
  class="emotion-1"
  role="switch"
>
  <span
    class="emotion-0"
  />
</button>
`;
